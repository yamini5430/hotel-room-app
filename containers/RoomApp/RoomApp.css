button.submitDetails {
    margin: 25px;
    width: 75px;
    height: 43px;
    background-color: #A9A9A9;
    font-size: 14px;
    cursor: pointer;
}

.roomList {
    display: flex;
    align-items: center;
    flex-flow: row;
    justify-content: space-between;
    width: 40%;
    margin: 25px;
}  