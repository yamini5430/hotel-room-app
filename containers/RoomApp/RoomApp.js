import React, {Component} from 'react';
import RoomList from '../../components/RoomList/RoomList';

const roomArr = ['room1','room2', 'room3', 'room4'];

class RoomApp extends Component {
   constructor(props){
       super(props);
        this.state  = {
            roomData : {}
        };
        this.handleSubmit = this.handleSubmit.bind(this);

   }

   getRoomDefaultData =()=> {
      var newObj = roomArr.reduce(function(obj,v,i){
        obj[v] = {
            enabled : i > 0 ? false : true,
            key : v+"-key",
            name : "Room " + (i+1),
            data : {
                "adultCount" : 1,
                "childrenCount" : 0	
            }
        };
        return obj;
    }, {});
    return newObj;
   }

   componentDidMount(){
        var data = {};
        if(localStorage && localStorage.getItem('roomData')){
            data = JSON.parse(localStorage.getItem('roomData'));
        } else {
            data = this.getRoomDefaultData();
        }

        this.setState({
            roomData : data 
        }); 
    }

   handleSubmit(){
       localStorage.setItem('roomData',JSON.stringify(this.state.roomData));
       alert("Hi User, Your data has been recorded!");
   }

   setRoomStatus = (roomArrData,status) => {
    let updatedState = {...this.state.roomData};
    roomArrData.forEach((v,i) => {
        let roomObj = {...this.state.roomData[v]};
        roomObj["enabled"] = status;
        roomObj["data"] = {adultCount : 1 , childrenCount : 0}
        updatedState[v] = roomObj;
    });
    this.setState({
        roomData : updatedState
    });

   }

   checkChangeHandler = (key,event) => {
    let checked = event.target.checked;
    let newRoomArr = [...roomArr];
    let tempState = {...this.state.roomData};
    newRoomArr.shift();
    if(checked){
      let updatedArr =  newRoomArr.filter(function(v){
            return ((newRoomArr.indexOf(v) <= newRoomArr.indexOf(key)) && !tempState[v]["enabled"])
        });
        this.setRoomStatus(updatedArr,true )

    } else {
        let updatedArr =  newRoomArr.filter(function(v){
            return (newRoomArr.indexOf(v) >= newRoomArr.indexOf(key) && tempState[v]["enabled"] );
        });
        this.setRoomStatus(updatedArr, false )
    }

   }

   updatePersonHandler = (event,type,comp) => {
    let selectValue = event.target.value;
    let updatedState =  {...this.state.roomData};
    let updatedRoomObj = {...this.state.roomData[type]};
    updatedRoomObj["data"][comp] = selectValue;
    updatedState[type] = updatedRoomObj;
    this.setState({
        roomData : updatedState
    })
   }

   render(){
       return (
         <div className = "roomApp">
             <RoomList data = {this.state.roomData} checkChangeHandler = {this.checkChangeHandler} updatePersonHandler =
             {this.updatePersonHandler} /> 
             <button className = "submitDetails" onClick = {this.handleSubmit}>Submit</button>
             
           </div>
       )
   }

}

export default RoomApp;