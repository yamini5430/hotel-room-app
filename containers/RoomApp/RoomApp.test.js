import React from 'react';
import ReactDOM from 'react-dom';
import RoomApp from './RoomApp';
import renderer from 'react-test-renderer';

// Snapshot for RoomApp React Component
test(" snapshot test", ()=>{
    const component = renderer.create(
         <RoomApp />
         );
         let tree = component.toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
})

