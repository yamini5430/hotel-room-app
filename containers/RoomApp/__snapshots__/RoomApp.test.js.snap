// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` snapshot test 1`] = `
<div
  className="roomApp"
>
  <div
    className="roomList"
  >
    <div
      className="room"
    >
      <div
        className="title"
      >
        Room 1
      </div>
      <div
        className="room-data"
      >
        <div
          className="adult-count"
        >
          <div
            className=""
          >
            <p>
              Adults
            </p>
            <p>
              (18+)
            </p>
          </div>
          <select
            disabled={false}
            onChange={[Function]}
            value={1}
          >
            <option
              value={1}
            >
              1
            </option>
            <option
              value={2}
            >
              2
            </option>
          </select>
        </div>
        <div
          className="children-count"
        >
          <div
            className=""
          >
            <p>
              Children
            </p>
            <p>
              (0-17)
            </p>
          </div>
          <select
            disabled={false}
            onChange={[Function]}
            value={0}
          >
            <option
              value={0}
            >
              0
            </option>
            <option
              value={1}
            >
              1
            </option>
            <option
              value={2}
            >
              2
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="room disabled"
    >
      <div
        className="checkbox"
      >
        <label>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
            value="Room 2"
          />
          Room 2
        </label>
      </div>
      <div
        className="room-data"
      >
        <div
          className="adult-count"
        >
          <div
            className=""
          >
            <p>
              Adults
            </p>
            <p>
              (18+)
            </p>
          </div>
          <select
            disabled={true}
            onChange={[Function]}
            value={1}
          >
            <option
              value={1}
            >
              1
            </option>
            <option
              value={2}
            >
              2
            </option>
          </select>
        </div>
        <div
          className="children-count"
        >
          <div
            className=""
          >
            <p>
              Children
            </p>
            <p>
              (0-17)
            </p>
          </div>
          <select
            disabled={true}
            onChange={[Function]}
            value={0}
          >
            <option
              value={0}
            >
              0
            </option>
            <option
              value={1}
            >
              1
            </option>
            <option
              value={2}
            >
              2
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="room disabled"
    >
      <div
        className="checkbox"
      >
        <label>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
            value="Room 3"
          />
          Room 3
        </label>
      </div>
      <div
        className="room-data"
      >
        <div
          className="adult-count"
        >
          <div
            className=""
          >
            <p>
              Adults
            </p>
            <p>
              (18+)
            </p>
          </div>
          <select
            disabled={true}
            onChange={[Function]}
            value={1}
          >
            <option
              value={1}
            >
              1
            </option>
            <option
              value={2}
            >
              2
            </option>
          </select>
        </div>
        <div
          className="children-count"
        >
          <div
            className=""
          >
            <p>
              Children
            </p>
            <p>
              (0-17)
            </p>
          </div>
          <select
            disabled={true}
            onChange={[Function]}
            value={0}
          >
            <option
              value={0}
            >
              0
            </option>
            <option
              value={1}
            >
              1
            </option>
            <option
              value={2}
            >
              2
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="room disabled"
    >
      <div
        className="checkbox"
      >
        <label>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
            value="Room 4"
          />
          Room 4
        </label>
      </div>
      <div
        className="room-data"
      >
        <div
          className="adult-count"
        >
          <div
            className=""
          >
            <p>
              Adults
            </p>
            <p>
              (18+)
            </p>
          </div>
          <select
            disabled={true}
            onChange={[Function]}
            value={1}
          >
            <option
              value={1}
            >
              1
            </option>
            <option
              value={2}
            >
              2
            </option>
          </select>
        </div>
        <div
          className="children-count"
        >
          <div
            className=""
          >
            <p>
              Children
            </p>
            <p>
              (0-17)
            </p>
          </div>
          <select
            disabled={true}
            onChange={[Function]}
            value={0}
          >
            <option
              value={0}
            >
              0
            </option>
            <option
              value={1}
            >
              1
            </option>
            <option
              value={2}
            >
              2
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <button
    className="submitDetails"
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;

exports[`>>>Room List--- Snapshot +++capturing Snapshot of Room List 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RoomList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "roomList",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "roomList",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
